pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -444,6 +445,65 @@ func getAddressByName(client *gophercloud.ServiceClient, name types.NodeName) (s
 	return addrs[0].Address, nil
 }
 
+// getInterfacesByName returns the node interfaces of the specified instance.
+func getInterfacesByName(client *gophercloud.ServiceClient, name types.NodeName) ([]attachinterfaces.Interface, error) {
+	var interfaces []attachinterfaces.Interface
+
+	// get server
+	srv, err := getServerByName(client, name)
+	if err != nil {
+		if err == ErrNotFound {
+			return interfaces, cloudprovider.InstanceNotFound
+		}
+		return interfaces, err
+	}
+
+	// get interfaces of server
+	serviceID := srv.ID
+	pager := attachinterfaces.List(client, serviceID)
+
+	interfaceList := make([]attachinterfaces.Interface, 0, 1)
+
+	err = pager.EachPage(func(page pagination.Page) (bool, error) {
+		s, err := attachinterfaces.ExtractInterfaces(page)
+		if err != nil {
+			return false, err
+		}
+		interfaceList = append(interfaceList, s...)
+		return true, nil
+	})
+	if err != nil {
+		return interfaces, err
+	}
+
+	// return ErrNotFound when server contains no interface
+	if len(interfaceList) == 0 {
+		return interfaces, ErrNotFound
+	}
+	return interfaces, nil
```

Comment:
@dims Yes, I update my PR to make it simpler, and have removed interfaceList.
Please add comment into latest PR if you have any questions, thank you.